src/client/AM.PA.MonitoringTool/SlackTracker/Daemon.cs

```diff
@@ -0,0 +1,345 @@
+
+                DatabaseConnector.SetAnalyzedDay(date);
+            }
+        }
+
+        private void GetLogs(DateTimeOffset latestSync)
+        {
+
+            List<LogData> logs = SlackConnector.GetLogs(latestSync);
+
+            if(logs == null || logs.Count == 0) { return;}
+
+            DatabaseConnector.SaveLogs(logs);
+        }
+
+        private void GetUsers(DateTimeOffset _latestSync)
+        {
+            List<User> users = SlackConnector.GetUsers();
+
+            DatabaseConnector.SaveUsers(users);
+        }
+
+        private void GetChannels(DateTimeOffset _latestSync)
+        {
+            List<Channel> channels = SlackConnector.GetChannels();
+
+            DatabaseConnector.SaveChannels(channels);
+        }
+
+        //Creates a timer that is used to periodically pull data from the slack API
+        private void CreateSlackPullTimer()
+        {
+            _slackTimer = new Timer();
+            _slackTimer.Elapsed += OnPullFromSlack;
+            _slackTimer.Interval = Settings.SYNCHRONIZE_INTERVAL;
+            _slackTimer.Enabled = true;
+        }
+        public void ChangeEnabledState(bool? slackTrackerEnabled)
+        {
+            Console.WriteLine(Settings.TRACKER_NAME + " is now " + (slackTrackerEnabled.Value ? "enabled" : "disabled"));
+            Database.GetInstance().SetSettings(Settings.TRACKER_ENABLED_SETTING, slackTrackerEnabled.Value);
+            Database.GetInstance().LogInfo("The participant updated the setting '" + Settings.TRACKER_ENABLED_SETTING + "' to " + slackTrackerEnabled.Value);
+
+            if (slackTrackerEnabled.Value && !_isPApaused)
+            {
+                CreateDatabaseTablesIfNotExist();
+                InternalStart();
+            }
+            else if (!slackTrackerEnabled.Value && !_isPApaused && IsRunning)
+            {
+                InternalStop();
+            }
+            else
+            {
+                Logger.WriteToConsole("Don't do anything, tracker is paused");
+            }
+        }
+
+        void OnTrackerDisabled()
+        {
+            IsRunning = false;
+            Database.GetInstance().SetSettings(Settings.TRACKER_ENABLED_SETTING, false);
+        }
+
+        //Called when new tokens were received from slack
+        private void Browser_RegistrationTokenEvent(string token)
+        {
+            CheckIfSecretsAreAvailable();
+            SlackConnector.GetAccessToken(token);
+        }
+
+        private void Browser_ErrorEvent()
+        {
+            throw new NotImplementedException();
+        }
+
+        //Called when the browser window used to retrieve tokens from slack should be closed
+        private void Browser_FinishEvent()
+        {
+            Application.Current.Dispatcher.Invoke((Action)(() =>
+            {
+                _browserWindow.Close();
+            }));
+        }
+
+        public override List<IVisualization> GetVisualizationsDay(DateTimeOffset date)
```

Comment:
can you please move the methods that implement the ITracker interface into the same region (i.e. #region Daemon)? That way it's more consistent with the other trackers; e.g. see WindowsActivityTracker.